### PR TITLE
Add context to Elasticsearch calls

### PR DESCRIPTION
**Pull request by mrodm**
Leverage the usage of contexts in some Elasticsearch calls found in benchrunner and testrunner.

### CODE REVIEW

File: internal/benchrunner/runners/system/runner.go [hunk 1]
```diff
@@ -873,7 +878,10 @@ func (r *runner) bulkMetrics(indexName string, sr searchResponse) error {
 
 	logger.Debugf("bulk request of %d events...", len(sr.Hits))
 
-	resp, err := r.options.ESMetricsAPI.Bulk(strings.NewReader(bulkBodyBuilder.String()))
+	resp, err := r.options.ESMetricsAPI.Bulk(
+		strings.NewReader(bulkBodyBuilder.String()),
```
Comment (jsoriano) [inline internal/benchrunner/runners/system/runner.go:882]: Nit.
```suggestion
	resp, err := r.options.ESMetricsAPI.Bulk(strings.NewReader(bulkBodyBuilder.String()),
```

File: internal/elasticsearch/ingest/pipeline.go [hunk 2]
```diff
@@ -86,7 +87,7 @@ func SimulatePipeline(api *elasticsearch.API, pipelineName string, events []json
 		return nil, fmt.Errorf("marshalling simulate request failed: %w", err)
 	}
 
-	r, err := api.Ingest.Simulate(bytes.NewReader(requestBody), func(request *elasticsearch.IngestSimulateRequest) {
+	r, err := api.Ingest.Simulate(bytes.NewReader(requestBody), api.Ingest.Simulate.WithContext(ctx), func(request *elasticsearch.IngestSimulateRequest) {
 		request.PipelineID = pipelineName
 	})
```
Comment (jsoriano) [inline internal/elasticsearch/ingest/pipeline.go:92]:
```suggestion
	r, err := api.Ingest.Simulate(bytes.NewReader(requestBody),
		api.Ingest.Simulate.WithContext(ctx),
		api.Ingest.Simulate.WithPipelineID(pipelineName),
	})
```

**Review by jsoriano — APPROVED** [review]
Nice, thanks!

File: internal/elasticsearch/ingest/pipeline.go (hunk 2 above)
Comment (mrodm) [reply internal/elasticsearch/ingest/pipeline.go:92]: Oh! I didn't check if it existed that function, thanks!

**Review by jsoriano — APPROVED** [review]

**Pull request merged**